src/openforms/formio/api/validators.py

```diff
@@ -65,6 +65,17 @@ def __call__(self, value: UploadedFile) -> None:
             whole_file = head + value.read()
             mime_type = magic.from_buffer(whole_file, mime=True)
 
+        # gh #4658
+        # Windows use application/x-zip-compressed as a mimetype for .zip files, which
+        # is deprecated but still we need to support it. Instead, the common case for
+        # zip files is application/zip or application/zip-compressed mimetype.
+        if (
+            value.content_type
+            in ["application/x-zip-compressed", "application/zip-compressed"]
+            and mime_type == "application/zip"
+        ):
+            value.content_type = "application/zip"
```

Comment:
why is this being assigned here? we are not supposed to mutate `value` inside a validator

Instead the condition should maybe be done similar like line 110?

```py
        elif mime_type == "application/zip" and value.content_type in (
            "application/zip-compressed",
            "application/x-zip-compressed",
        ):
            return
```